libs/motor/inc/public/tfc/motors/positioner.hpp

```diff
@@ -0,0 +1,335 @@
+
+  // https://github.com/PaulStoffregen/Encoder/blob/master/Encoder.h#L303
+  //                           _______         _______
+  //               Pin1 ______|       |_______|       |______ Pin1
+  // negative <---         _______         _______         __      --> positive
+  //               Pin2 __|       |_______|       |_______|   Pin2
+
+  //	new 	new	    old 	old
+  //	pin2	pin1	pin2	pin1	Result
+  //	----	----	----	----	------
+  //	0	0	0	0	no movement
+  //	0	0	0	1	+1
+  //	0	0	1	0	-1
+  //	0	0	1	1	+2  (assume pin1 edges only)
+  //	0	1	0	0	-1
+  //	0	1	0	1	no movement
+  //	0	1	1	0	-2  (assume pin1 edges only)
+  //	0	1	1	1	+1
+  //	1	0	0	0	+1
+  //	1	0	0	1	-2  (assume pin1 edges only)
+  //	1	0	1	0	no movement
+  //	1	0	1	1	-1
+  //	1	1	0	0	+2  (assume pin1 edges only)
+  //	1	1	0	1	-1
+  //	1	1	1	0	+1
+  //	1	1	1	1	no movement
+  /*
+          // Simple, easy-to-read "documentation" version :-)
+          //
+          void update(void) {
+                  uint8_t s = state & 3;
+                  if (digitalRead(pin1)) s |= 4;
+                  if (digitalRead(pin2)) s |= 8;
+                  switch (s) {
+                          case 0: case 5: case 10: case 15:
+                                  break;
+                          case 1: case 7: case 8: case 14:
+                                  position++; break;
+                          case 2: case 4: case 11: case 13:
+                                  position--; break;
+                          case 3: case 12:
+                                  position += 2; break;
+                          default:
+                                  position -= 2; break;
+                  }
+                  state = (s >> 2);
+          }
+  */
+
+  void first_tacho_update(bool first_new_val) noexcept {
+    if (buffer_.front().first_tacho_state) {
+      if (buffer_.front().second_tacho_state) {
+        if (!first_new_val) {
+          decrement();
+        }
+      } else {
+        if (!first_new_val) {
+          increment();
+        }
+      }
+    } else {
+      if (buffer_.front().second_tacho_state) {
+        if (first_new_val) {
+          increment();
+        }
+      } else {
+        if (first_new_val) {
+          decrement();
+        }
+      }
+    }
+
+    auto now{ time_point_t::clock::now() };
+    buffer_.emplace(first_new_val, buffer_.front().second_tacho_state, now);
+  }
+
+  void second_tacho_update(bool second_new_val) noexcept {
+    if (buffer_.front().first_tacho_state) {
+      if (buffer_.front().second_tacho_state) {
+        if (!second_new_val) {
+          increment();
+        }
+      } else {
+        if (second_new_val) {
+          decrement();
+        }
+      }
+    } else {
+      if (!buffer_.front().second_tacho_state) {
+        if (second_new_val) {
+          increment();
+        }
+      } else {
+        if (!second_new_val) {
+          decrement();
+        }
+      }
+    }
```

Comment:
This is hard to read

Reply:
Can we not merge the data and utilize a similar function as the update function in the comments. It is also allot faster.